aten/src/ATen/native/Loss.cpp

```diff
@@ -79,10 +79,26 @@ Tensor hinge_embedding_loss(const Tensor& self, const Tensor& target, double mar
 
 Tensor triplet_margin_loss(const Tensor& anchor, const Tensor& positive, const Tensor& negative, double margin,
                            double p, double eps, bool swap, int64_t reduction) {
-  auto dist_pos = at::pairwise_distance(anchor, positive, p, eps);
-  auto dist_neg = at::pairwise_distance(anchor, negative, p, eps);
+  auto a_dim = anchor.dim();
+  auto p_dim = positive.dim();
+  auto n_dim = negative.dim();
+  TORCH_CHECK(
+      a_dim == p_dim && p_dim == n_dim,
+      "All inputs should have same dimension but got ",
+      a_dim,
+      "D, ",
+      p_dim,
+      "D and ",
+      n_dim,
+      "D inputs.")
+  auto pairwise_distance = [p, eps](const Tensor& x1, const Tensor& x2) {
+    auto innermost_dim = x1.dim() - 1;
+    return at::norm(x1 - x2 + eps, p, innermost_dim, false);
+  };
```

Comment:
Hm, I'm now rethinking my original decision to leave out `nn.PairwiseDistance` from supporting no batch dim. There's nothing that prevents it from being updated as you did here to support 1D inputs - that would simplify the logic here.

I updated the [rollup issue](https://github.com/pytorch/pytorch/issues/60585) to move `nn.PairwiseDistance` into the list of things to be updated.

Reply:
Done. Thanks :)!